apps/ejabberd/src/mod_mam_odbc_prefs.erl

```diff
@@ -133,20 +133,46 @@ set_prefs(_Result, Host, UserID, _ArcJID, DefaultMode, AlwaysJIDs, NeverJIDs) ->
         {error, Error}
     end.
 
+
+order_by_in_delete(Host) ->
+    case ejabberd_odbc:db_engine(Host) of
+        mysql ->
```

Comment:
Only mysql has support for this.
It orders locks to avoid dead locks in theory (in practice we still have them and should retry).